File: frontend/src/services/unifiedApprovalService.js
import apiClient from './api';

export const getPendingCounts = () => apiClient.get('/suite/approvals/counts');
export const getDelegations = () => apiClient.get('/suite/approvals/delegations');
export const delegateApproval = (dto) => apiClient.post('/suite/approvals/delegate', dto);

File: backend/src/main/java/com/awais/hr/module/approvals/service/UnifiedApprovalService.java
package com.awais.hr.module.approvals.service;

import java.util.List;
import java.util.Map;

public interface UnifiedApprovalService {
    Map<String, Object> getPendingCounts();
    List<Map<String, Object>> getDelegations();
    Map<String, Object> delegateApproval(Map<String, Object> body);
}
